### PR TITLE
feat(docs): add docs guide for building themes

### LINKED ISSUE #15641: [themes] Write a guide for authoring themes
# Description

Write a guide for authoring themes. This will be a guide complement to the fuller tutorial (captured in #14099). This will be based around ~using the theme jam submission starter, and later be updated when we have an official workspace starter~ the official workspace starter.

## Related

#14099

### PR DESCRIPTION
**Pull request by amberleyromo**
## Description

Introduces a docs guide for building themes, oriented around using the workspace starter.

## Notes

- This references the fuller tutorial -- that PR (#15668) should be merged first.

## Related Issues

Closes #15641.

Related to #15668.

### CODE REVIEW

**Comment by sidharthachatterjee** [comment]
Blocked by https://github.com/gatsbyjs/gatsby/pull/15668

File: docs/docs/themes/building-themes.md [hunk 1]
```diff
@@ -2,15 +2,94 @@
...
+├── package.json
+├── README.md
+└── yarn.lock
+```
+
+Yarn workspaces are a great way to set up a project for theme development because they support housing multiple packages in a single parent directory and link them together.
+
+For Gatsby theme development, that means you can keep multiple themes and example sites together in a single project, and develop them locally.
+
+> 💡 If you prefer, using `yarn link` or `npm link` are viable alternatives. In general, we prefer the yarn workspaces approach, and that's what the starter and this guide will reflect.
```
Comment (marcysutton) [inline docs/docs/themes/building-themes.md]:
```suggestion
> 💡 If you prefer, using `yarn link` or `npm link` are viable alternatives. In general, Gatsby recommends the yarn workspaces approach for building themes, and that's what the starter and this guide will reflect.
```

File: docs/docs/themes/building-themes.md [hunk 2]
```diff
@@ -2,15 +2,94 @@
...
+
+### `/gatsby-theme-minimal`
+
+The `/gatsby-theme-minimal` directory is the starting point of the new theme you'll develop.
+
+Inside it you'll find:
+
+- `gatsby-config.js`: An empty gatsby-config that you can use as a starting point for building functionality into your theme.
+- `index.js`: Since themes also function as plugins, this is an empty file that Gatsby requires in order to use this theme as a plugin.
+- `package.json`: The dependencies that your theme will pull in when people install it. Gatsby should be a peer dependency.
```
Comment (marcysutton) [inline docs/docs/themes/building-themes.md]:
```suggestion
- `package.json`: A file listing the dependencies that your theme will pull in when people install it. Gatsby should be a peer dependency.
```

File: docs/docs/themes/building-themes.md [hunk 3]
```diff
@@ -2,15 +2,94 @@
...
+
+### `/example`
+
+The `/example` directory is an example Gatsby site that installs and uses the local theme, `gatsby-theme-minimal`.
+
+Inside it you'll find:
+
+- `gatsby-config.js`: Specifies which theme to use and any other one-off config a site might need.
```
Comment (marcysutton) [inline docs/docs/themes/building-themes.md]:
```suggestion
- `gatsby-config.js`: Specifies which theme to use and any other one-off configuration a site might need.
```

File: docs/docs/themes/building-themes.md [hunk 4]
```diff
@@ -2,15 +2,94 @@
...
+- `/src`: Contains source code such as one-off pages or components that might live in a user's site.
```
Comment (marcysutton) [inline docs/docs/themes/building-themes.md]:
```suggestion
- `/src`: Contains source code such as custom pages or components that might live in a user's site.
```

File: docs/docs/themes/building-themes.md [hunk 5]
```diff
@@ -2,15 +2,94 @@
...
+## Further resources
+
+### Gatsby Theme Authoring (Video course)
 
 Watch the new [Egghead course on Authoring Gatsby Themes](https://egghead.io/courses/gatsby-theme-authoring).
 
-## Read through source code
+### Building a Gatsby Theme (Tutorial)
+
+Check out the [tutorial for building a Gatsby theme](/tutorial/building-a-theme). The tutorial goes into much more detail than this g This is written as a companion to the [Egghead theme authoring course](https://egghead.io/courses/gatsby-theme-authoring), so they can be used together!
```
Comment (marcysutton) [inline docs/docs/themes/building-themes.md]:
```suggestion
Check out the [tutorial for building a Gatsby theme](/tutorial/building-a-theme). The step-by-step tutorial goes into much more detail than this docs guide. It was written as a companion to the [Egghead theme authoring course](https://egghead.io/courses/gatsby-theme-authoring), so they can be used together!
```

**Pull request merged**